lib/routes/routeBackbeat.js

```diff
@@ -502,7 +529,10 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
                 pushReplicationMetric(objMd, omVal, bucketName, objectKey, log);
                 if (objMd &&
                     headers['x-scal-replication-content'] !== 'METADATA' &&
-                    versionId &&
+                    // - if a version ID is provided in the query string (updating MD of a specific version) OR
+                    // - if no version ID provided and versioning is not enabled (updating MD of the master key),
+                    // data locations are allowed to be removed as they will be overwritten.
+                    (versionId || !versioning) &&
```

Comment:
I have a doubt that we may remove valid data in the following case, not completely sure, to be checked:

- versioning-suspended bucket
- no existing null version

I am thinking that if `objMd` is the metadata of the current master (a real version), we may remove the master's data and create a new null version, making the old master unreadable.

Reply:
If there is no existing null version, the object MD (objMd) of the null version will be undefined, resulting in the condition being false.